recipes/libpqxx/all/conanfile.py

```diff
@@ -0,0 +1,134 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment, CMake
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class LibpqxxRecipe(ConanFile):
+    name = "libpqxx"
+    settings = "os", "compiler", "build_type", "arch"
+    description = "The official C++ client API for PostgreSQL"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jtv/libpqxx"
+    license = "BSD-3-Clause"
+    topics = ("conan", "libpqxx", "postgres", "postgresql", "database", "db")
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = "libpq/11.5"
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _using_cmake(self):
+        return self.settings.os == "Windows"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "6",
+            "apple-clang": "10"
+        }
+
+        if compiler in minimal_version and \
+           compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports"
+                                            " at least C++17. %s %s is not"
+                                            " supported." % (self.name, compiler, compiler_version))
+        minimal_cpp_standard = "17"
+        supported_cppstd = ["17", "20"]
+
+        if not self.settings.compiler.cppstd:
+            self.output.info("Settings c++ standard to {}".format(minimal_cpp_standard))
+            self.settings.compiler.cppstd = minimal_cpp_standard
```

Comment:
@uilianries recently told me about`tools.check_min_cppstd`.
Because modifying `self.settings` is not really wanted.
You can also do a `self.output.warn` to at least notify the developer.

Reply:
Sure.

Remove this code.